### PR TITLE
MAILET-119 RemoteDelivery should not rely on error message for retires

### PR DESCRIPTION
**Pull request by chibenwa**
It should rather use a dedicated mail attribute.

However, I strictly don't know how to test this change...

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java [hunk 1]
```diff
@@ -856,6 +853,14 @@ public void run() {
         }
     }
 
+    private int retrieveRetries(Mail mail) {
+        try {
+            return Optional.fromNullable((Integer) mail.getAttribute(RETRY_ATTRIBUTE_KEY)).or(0);
+        } catch(ClassCastException e) {
+            return 0;
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: should be log, don't you think ?

Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java]: Ok

**Comment by chibenwa** [comment]
Do you have any idea on "How to test this beast ?". No test is Ok here ?

**Comment by mbaechler** [comment]
I would start by extracting the delivery logic to another class. That way you could probably test the retry mechanism by injecting a mock for delivery service. WDYT ?

**Pull request closed without merging**